openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMModelCreator.java

```diff
@@ -352,4 +344,43 @@ private static boolean schemaHasStringFields(final DatasetSchema schema) {
                 .map(fieldName -> fieldName.replace(" ", "_"))
                 .toArray(String[]::new);
     }
+
+    /**
+     * Performs a one-by-one feature name comparison between a
+     * {@link DatasetSchema} and an array of feature names.
+     * This way the first mismatch is logged, improving debug.
+     *
+     * @param schema Schema
+     * @param featureNames Feature names to validate.
+     * @return {@code true} if the schema predictive field names
+     * match the provided array, {@code false} otherwise.
+     * @since 1.0.18
+     */
+    private boolean schemaMatchAllFeatures(final DatasetSchema schema, final String[] featureNames) {
+
+        final String[] schemaFeatureNames = getFeatureNamesFrom(schema);
+
+        boolean isMatch = true;
+
+        for (int i = 0; i < featureNames.length; i++) {
+
+            if (!schemaFeatureNames[i].equals(featureNames[i])) {
+
+                logger.error("Schema with wrong predictive field name at index {}: '{}' Expected: '{}'",
```

Comment:
Maybe this one should be logger.info? I understand your point back then.

The error is only one, we only detail the several issues in that single error which is raised.

If we have info it always shows up by default even if the users don't set to debug level right?